numpy/_core/src/common/simd/lsx/arithmetic.h

```diff
@@ -0,0 +1,311 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+#ifndef _NPY_SIMD_LSX_ARITHMETIC_H
+#define _NPY_SIMD_LSX_ARITHMETIC_H
+
+/***************************
+ * Addition
+ ***************************/
+// non-saturated
+#define npyv_add_u8  __lsx_vadd_b
+#define npyv_add_s8  __lsx_vadd_b
+#define npyv_add_u16 __lsx_vadd_h
+#define npyv_add_s16 __lsx_vadd_h
+#define npyv_add_u32 __lsx_vadd_w
+#define npyv_add_s32 __lsx_vadd_w
+#define npyv_add_u64 __lsx_vadd_d
+#define npyv_add_s64 __lsx_vadd_d
+#define npyv_add_f32 __lsx_vfadd_s
+#define npyv_add_f64 __lsx_vfadd_d
+
+// saturated
+#define npyv_adds_u8  __lsx_vsadd_bu
+#define npyv_adds_s8  __lsx_vsadd_b
+#define npyv_adds_u16 __lsx_vsadd_hu
+#define npyv_adds_s16 __lsx_vsadd_h
+#define npyv_adds_u32 __lsx_vsadd_wu
+#define npyv_adds_s32 __lsx_vsadd_w
+#define npyv_adds_u64 __lsx_vsadd_du
+#define npyv_adds_s64 __lsx_vsadd_d
+
+
+/***************************
+ * Subtraction
+ ***************************/
+// non-saturated
+#define npyv_sub_u8  __lsx_vsub_b
+#define npyv_sub_s8  __lsx_vsub_b
+#define npyv_sub_u16 __lsx_vsub_h
+#define npyv_sub_s16 __lsx_vsub_h
+#define npyv_sub_u32 __lsx_vsub_w
+#define npyv_sub_s32 __lsx_vsub_w
+#define npyv_sub_u64 __lsx_vsub_d
+#define npyv_sub_s64 __lsx_vsub_d
+#define npyv_sub_f32 __lsx_vfsub_s
+#define npyv_sub_f64 __lsx_vfsub_d
+
+// saturated
+#define npyv_subs_u8  __lsx_vssub_bu
+#define npyv_subs_s8  __lsx_vssub_b
+#define npyv_subs_u16 __lsx_vssub_hu
+#define npyv_subs_s16 __lsx_vssub_h
+#define npyv_subs_u32 __lsx_vssub_wu
+#define npyv_subs_s32 __lsx_vssub_w
+#define npyv_subs_u64 __lsx_vssub_du
+#define npyv_subs_s64 __lsx_vssub_d
+
+
+/***************************
+ * Multiplication
+ ***************************/
+// non-saturated
+#define npyv_mul_u8  __lsx_vmul_b
+#define npyv_mul_s8  __lsx_vmul_b
+#define npyv_mul_u16 __lsx_vmul_h
+#define npyv_mul_s16 __lsx_vmul_h
+#define npyv_mul_u32 __lsx_vmul_w
+#define npyv_mul_s32 __lsx_vmul_w
+#define npyv_mul_f32 __lsx_vfmul_s
+#define npyv_mul_f64 __lsx_vfmul_d
+
+
+/***************************
+ * Integer Division
+ ***************************/
+// See simd/intdiv.h for more clarification
+// divide each unsigned 8-bit element by a precomputed divisor
+NPY_FINLINE npyv_u8 npyv_divc_u8(npyv_u8 a, const npyv_u8x3 divisor)
+{
+    const __m128i bmask = __lsx_vreplgr2vr_w(0x00FF00FF);
+    const __m128i shf1b = __lsx_vreplgr2vr_b(0xFFU >> __lsx_vpickve2gr_w(divisor.val[1], 0));
+    const __m128i shf2b = __lsx_vreplgr2vr_b(0xFFU >> __lsx_vpickve2gr_w(divisor.val[2], 0));
+    // high part of unsigned multiplication
+    __m128i mulhi_even  = __lsx_vmul_h(__lsx_vand_v(a, bmask), divisor.val[0]);
+    __m128i mulhi_odd   = __lsx_vmul_h(__lsx_vsrli_h(a, 8), divisor.val[0]);
+            mulhi_even  = __lsx_vsrli_h(mulhi_even, 8);
+    __m128i mulhi       = npyv_select_u8(bmask, mulhi_even, mulhi_odd);
```

Comment:
```suggestion
    __m128i mulhi       = __lsx_vmuh_b(a, divisor.val[0]);
```
Is there any reason for emulate it rather than using the native  `__lsx_vmuh_b`?

Reply:
Byte type data is not applicable, it will cause the functional test to fail.